### PR TITLE
[NCL-5797] Add support for the LATEST on /group-configs/{id}/group-bu…

### PR DESCRIPTION
**Pull request by akridI**
…ilds

### Checklist:

* [ ] Have you added unit tests for your change?

### CODE REVIEW

File: integration-test/src/test/java/org/jboss/pnc/integration/endpoints/GroupConfigurationEndpointTest.java [hunk 1]
```diff
@@ -31,15 +31,11 @@
...
 import org.jboss.pnc.client.patch.PatchBuilderException;
 import org.jboss.pnc.common.concurrent.MDCExecutors;
-import org.jboss.pnc.dto.BuildConfiguration;
-import org.jboss.pnc.dto.BuildConfigurationRef;
-import org.jboss.pnc.dto.GroupConfiguration;
-import org.jboss.pnc.dto.Product;
-import org.jboss.pnc.dto.ProductRef;
-import org.jboss.pnc.dto.ProductVersion;
-import org.jboss.pnc.dto.ProductVersionRef;
+import org.jboss.pnc.dto.*;
```
Comment (michalovjan) [inline integration-test/src/test/java/org/jboss/pnc/integration/endpoints/GroupConfigurationEndpointTest.java:34]: single class imports

File: rest/src/main/java/org/jboss/pnc/rest/endpoints/GroupConfigurationEndpointImpl.java [hunk 2]
```diff
@@ -26,12 +26,10 @@
 import org.jboss.pnc.dto.requests.GroupBuildRequest;
 import org.jboss.pnc.dto.response.Page;
 import org.jboss.pnc.facade.BuildTriggerer;
-import org.jboss.pnc.facade.providers.api.BuildConfigurationProvider;
-import org.jboss.pnc.facade.providers.api.BuildProvider;
-import org.jboss.pnc.facade.providers.api.GroupBuildProvider;
-import org.jboss.pnc.facade.providers.api.GroupConfigurationProvider;
+import org.jboss.pnc.facade.providers.api.*;
```
[the remark this answers is not in the document]
Comment (akridI) [reply rest/src/main/java/org/jboss/pnc/rest/endpoints/GroupConfigurationEndpointImpl.java]: ah, IDEA defaults :smile:

File: facade/src/main/java/org/jboss/pnc/facade/providers/GroupBuildProviderImpl.java [hunk 3]
```diff
@@ -130,18 +142,24 @@ private Consumer<Result> notifyOnDeletionCompletion(String callbackUrl) {
...
-            String q,
-            String groupConfigurationId) {
-        return queryForCollection(
-                pageIndex,
-                pageSize,
-                sort,
-                q,
-                withBuildConfigSetId(Integer.valueOf(groupConfigurationId)));
+    public Page<GroupBuild> getGroupBuilds(GroupBuildPageInfo groupBuildPageInfo) {
+        Integer configSetId = groupConfigurationMapper.getIdMapper()
```
[the remark this answers is not in the document]
Comment (akridI) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GroupBuildProviderImpl.java]: good catch @michalovjan

File: rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/GroupBuildsFilterParameters.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ *
+ * @author Adam Krídl &lt;akridl@redhat.com&gt;
+ */
+@Data
+public class GroupBuildsFilterParameters {
+
+    /**
+     * {@value SwaggerConstants#LATEST_BUILD_DESC}
+     */
+    @Parameter(description = SwaggerConstants.LATEST_BUILD_DESC)
```
Comment (janinko) [inline rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/GroupBuildsFilterParameters.java:38]: this description is misleading. should be talking about group builds

**Review by janinko — CHANGES_REQUESTED** [review]

File: rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/GroupBuildsFilterParameters.java (hunk 4 above)
Comment (akridI) [reply rest-api/src/main/java/org/jboss/pnc/rest/api/parameters/GroupBuildsFilterParameters.java]: changed, see #27aeb4e

File: facade/src/main/java/org/jboss/pnc/facade/providers/api/GroupBuildPageInfo.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ */
+@Data
+@Builder
+public class GroupBuildPageInfo {
+
+    private int pageIndex;
+    private int pageSize;
+    private String sort;
+    private String q;
+    private String groupConfigurationId;
```
[the remark this answers is not in the document]
Comment (akridI) [reply facade/src/main/java/org/jboss/pnc/facade/providers/api/GroupBuildPageInfo.java]: yeah, it's definitely more reusable this way, well done @janinko

File: facade/src/main/java/org/jboss/pnc/facade/providers/GroupBuildProviderImpl.java [hunk 6]
```diff
@@ -130,18 +142,23 @@ private Consumer<Result> notifyOnDeletionCompletion(String callbackUrl) {
...
-            String q,
-            String groupConfigurationId) {
-        return queryForCollection(
-                pageIndex,
-                pageSize,
-                sort,
-                q,
-                withBuildConfigSetId(Integer.valueOf(groupConfigurationId)));
+    public Page<GroupBuild> getGroupBuilds(GroupBuildPageInfo groupBuildPageInfo, String groupConfigId) {
+        Integer groupConfigIdModel = groupConfigurationMapper.getIdMapper().toEntity(groupConfigId);
```
Comment (akridI) [inline facade/src/main/java/org/jboss/pnc/facade/providers/GroupBuildProviderImpl.java:146]: was struggling a little bit with variable names in this part, so feel free to suggest better ones if it comes to your mind

Comment (janinko) [reply facade/src/main/java/org/jboss/pnc/facade/providers/GroupBuildProviderImpl.java:146]: it's fine

**Review by janinko — APPROVED** [review]

**Comment by janinko** [comment]
retest this please

**Pull request merged**